lib/actions.ts

```diff
@@ -2,15 +2,39 @@
 
 import * as db from "@/lib/db";
 import * as Schemas from "drizzle/schema";
+import { signIn } from "@/lib/auth";
 import { auth } from "./auth";
 import * as blob from "@vercel/blob";
-import { NewProjectFormSchema } from "./types";
+import { NewProjectFormSchema, RegisterSchemaType } from "./types";
 import { redirect } from "next/navigation";
 import { defaulTaskSpecs } from "./tasks";
 import assert from "assert";
 
 const VERCEL_BLOB_FAKE_FILES = true;
 
+export async function registerUser(data: RegisterSchemaType) {
+  try {
+    let user = await db.getUser(data.email);
+
+    if (user.length > 0) {
+      throw new Error("User already exists");
+    }
+
+    await db.createUser(data.name, data.phone ?? "", data.email, data.password);
+  } catch (error) {
+    throw error;
+  }
+}
+
+export async function signInFromLogin(data: any) {
```

Comment:
actually, this function isn't needed, you can call `signIn("credentials",`directly on the front end

Reply:
Not 100% if this is a good idea though, if you could look up some other examples. does nextauth validate the data itself? I assume so, but it would be good to double check